kv/src/main/kotlin/net/consensys/cava/kv/LevelDBKeyValueStore.kt

```diff
@@ -18,15 +18,20 @@ package net.consensys.cava.kv
 import kotlinx.coroutines.experimental.Unconfined
 import kotlinx.coroutines.experimental.withContext
 import net.consensys.cava.bytes.Bytes
-import org.iq80.leveldb.DB
+import org.fusesource.leveldbjni.JniDBFactory
+import org.iq80.leveldb.Options
+import java.nio.file.Path
 
 /**
  * A key-value store backed by LevelDB.
  */
-class LevelDBKeyValueStore(private val levelDB: DB) : KeyValueStore {
+class LevelDBKeyValueStore(databasePath: Path) : KeyValueStore {
+
+  private val db = JniDBFactory.factory.open(databasePath.toFile(),
+    Options().createIfMissing(true).cacheSize((100 * 1048576).toLong()))
```

Comment:
hardcoded cache size? Should probably be passed in (with a default). Are there other options one might want to set? If so, a builder might be appropriate...

Reply:
Yes, there are other options.
Options itself uses the builder approach. I can change the constructor to take an Options object.